src/api/newsletter.rs

```diff
@@ -40,13 +50,42 @@ pub async fn subscribe_handler(
     Ok(HttpResponse::NotImplemented().finish())
 }
 
+pub async fn subscribe_anonymous_handler(
+    basket: Data<Option<Basket>>,
+    subscription_req: web::Json<SubscriptionRequest>,
+) -> Result<HttpResponse, ApiError> {
+    if let Some(basket) = &**basket {
```

Comment:
Why do we need this check here, but not in `subscribe_handler()` in line 79?

Reply:
Here it's an `Option<Basket>` in `subscribe_handler()` it's `&Basket` as it's used internally.